frontend/src/pages/ImportCkan/SelectGroupsScreen.jsx

```diff
@@ -0,0 +1,589 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import { useSelector } from 'react-redux';
+
+import { Input } from '../../sharedComponents/Input';
+import { LoadingButton } from '../../sharedComponents/LoadingButton';
+
+// Joink
+import { RootLevelCategorySelector } from '../NewMetadataType/RootLevelCategorySelector';
+
+import Resource from './Resource';
+
+const OuterContainer = styled.div`
+`;
+
+const WizardContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-content: space-around;
+  max-width: 50em;
+  min-width: 20em;
+  padding: 1em;
+  margin: 10px 0px;
+  background-color: white;
+  border-radius: 4px;
+  -webkit-box-shadow: 0 0.0625em 0.125em rgba(0,0,0,0.15);
+  -moz-box-shadow: 0 0.0625em 0.125em rgba(0,0,0,0.15);
+  box-shadow: 0 0.0625em 0.125em rgba(0,0,0,0.15);
+`;
+
+const ErrorContainer = styled.div`
+  background-color: #ffcccc;
+  border-radius: 4px;
+  border: 1px solid #ff5555;
+  padding: 1em;
+`;
+
+const ProgressBar = styled.div`
+    height: 2em;
+    background-color: #aaffaa;
+`;
+
+const ProgressBarContainer = styled.div`
+    width: 100%;
+    position: relative;
+    >p {
+        position: absolute;
+        top: 0;
+    }
+`;
+
+const ResourceContainer = styled.div`
+
+>div {
+    border-top: 1px solid #000;
+}
+`;
+
+const UpcomingEntry = styled.div`
+    border-bottom: 1px solid #aaa;
+`;
+
+const NavigationContainer = styled.div`
+    display: flex;
+
+`;
+
+const Select = styled.select`
+  flex: 0 0 2em;
+  margin-bottom: 5px;
+  font-size: 16px;
+`;
+
+const TextArea = styled.textarea`
+  min-height: 150px;
+  width: 100%;
+  resize: vertical;
+  margin-bottom: 5px;
+  font-size: 16px;
+  border-radius: 4px;
+  color: black;
+  border: 1px solid darkgray;
+`;
+
+const OptionContainer = styled.div`
+    width: 100%;
+    border-radius: 0.4em;
+    display: flex;
+
+    background-color: #ccccff;
+    border: 1px solid #A19AE6;
+`;
+
+const OptionEntryContainer = styled.div`
+    flex: 1;
+    border-right: 1px solid #A19AE6;
+    padding: 1em 1em;
+
+    cursor: pointer;
+
+    :hover {
+        background-color: #A19AE6;
+    }
+
+    :last {
+        border-right: none;     
+    }
+`;
+
+const OptionEntry = ({ current, name, onClick }) => (
+  <OptionEntryContainer style={current ? { backgroundColor: '#A19AE6' } : {}} onClick={onClick}>
+    {name}
+  </OptionEntryContainer>
+);
+OptionEntry.propTypes = {
+  current: PropTypes.bool.isRequired,
+  name: PropTypes.string.isRequired,
+  onClick: PropTypes.func.isRequired,
+};
+
+
+export const SelectGroupsScreen = ({ groups, updateGroup, nextState }) => {
+  const [loading, setLoading] = useState(0);
+  const [saving, setSaving] = useState(false);
+  const [total, setTotal] = useState(0);
+  const [metadataTypes, setMetadataTypes] = useState({});
+  const [tags, setTags] = useState([]);
```

Comment:
we have the custom hook useGetTags in `frontend/src/sharedComponents/hooks`, which might cut down on duplicate code

Reply:
I am fetching tags at this stage on purpose, to assure i have an up-to-date tag list

(I use the list to determine what tags are safe to add to a metadata type)

Reply:
the useGetTags custom hook would also fetch tags (and have an up-to-date list) when the component mounts. I think that either I am misunderstanding what you need the fetch for or you are misunderstanding what the hook does